component/src/stake/component.rs

```diff
@@ -240,13 +244,10 @@ pub(crate) trait StakingImpl: StateWriteExt {
                 Ok(())
             }
             (Active | Inactive | Disabled | Jailed, Tombstoned) => {
-                let penalty = self
-                    .get_chain_params()
-                    .await?
-                    .slashing_penalty_misbehavior_bps;
+                let penalty = self.get_chain_params().await?.slashing_penalty_misbehavior;
 
-                // Apply the penalty to the validator's current exchange rate.
-                self.apply_slashing_penalty(identity_key, penalty).await?;
+                // Record the slashing penalty on this validator.
+                self.record_slashing_penalty(identity_key, penalty).await?;
```

Comment:
We no longer apply the slashing penalty to the validator's exchange rate immediately. Instead, we record the penalty when it happens, and apply the compounded penalty at the epoch boundary.